images/egress/router/egress-router.sh

```diff
@@ -130,6 +130,26 @@ function wait_until_killed() {
     wait
 }
 
+function gen_dnsproxy_iptables_rules() {
+    pod_ip=$(ip addr show dev eth0 | grep -Po 'inet \K[\d.]+')
+    if [[ -z "${pod_ip}" ]]; then
+        die "Failed to fetch Pod IP"
+    fi
+      
+    echo -A PREROUTING -p tcp -d "${pod_ip}" -j DNAT --to-destination "${EGRESS_SOURCE}"
```

Comment:
In the standard egress-router, iptables rules are set on the pod and packets from service IP to pod IP will hit the iptables on the pod that does the actual job. 
In this dns proxy case, haproxy is running on the macvlan interface and packets from service IP will only reach the pod IP. This additional rule will forward all tcp packets from pod to macvlan interface which has the egress source IP.